README.md

```diff
@@ -124,22 +124,124 @@ add('X');
 
 ### Types
 
-sanctuary-def defines `Type` values for JavaScript's built-in types:
+Conceptually, a type is a set of values. One can think of a value of
+type `Type` as a function of type `Any -> Boolean` (though this is an
```

Comment:
I might clarify this by adding something like `One can think of a value of type `Type` as a function of type `Any -> Boolean` that tests values for membership in this set.`